tools/tcpstates.py

```diff
@@ -129,8 +129,8 @@
             .oldstate = args->oldstate, .newstate = args->newstate};
         data4.skaddr = (u64)args->skaddr;
         data4.ts_us = bpf_ktime_get_ns() / 1000;
-        bpf_probe_read(&data4.saddr, sizeof(u32), args->saddr);
-        bpf_probe_read(&data4.daddr, sizeof(u32), args->daddr);
+        bpf_probe_read(&data4.saddr, sizeof(data4.saddr), args->saddr);
+        bpf_probe_read(&data4.daddr, sizeof(data4.daddr), args->daddr);
```

Comment:
Could you also remove the `bpf_probe_read` calls here? We can use `__buildin_memcpy` instead.

Reply:
Same below for IPv6.

Reply:
Ok, should we prefer to use __buildin_memcpy?